desmume/src/frontend/posix/shared/ctrlssdl.cpp

```diff
@@ -39,7 +41,8 @@ u16 nbr_joy;
 mouse_status mouse;
 static int fullscreen;
 
-static SDL_Joystick **open_joysticks = NULL;
+//List of currently connected joysticks (key - instance id, value - SDL_Joystick structure, joystick number)
+static std::unordered_map<SDL_JoystickID, std::pair<SDL_Joystick *, SDL_JoystickID> > open_joysticks;
```

Comment:
seems like overkill, currently we support a max of 16 joysticks, which means the joystick id we use fits into one byte... and the average gamer certainly has max 4 joysticks connected so everything would fit into one cache line using a dumb loop over an array. i'm pretty certain the overhead involved in the use of a map for such tiny amount of data is way slower

Reply:
> which means the joystick id we use fits into one byte

actually, this could be derived from the array index, so it's for free. we only need to make sure to fix the ordering of the array on insertion/removal of a device

Reply:
Then I probably just use static array of 16 elements?

Reply:
yeah, sounds like the best option. in the unlikely case that more than 16 joys are connected, we just have to discard those > 15.

Reply:
nbr_joy is used by the "interface" port. if you can add a tiny function that counts the joysticks in the array, the interface can call that and we can get read of the variable

Reply:
Added int get_number_of_joysticks()

Reply:
Should I also make corresponding changes to gtk2 frontend in my branch?

Reply:
yes, please

Reply:
Done